chain/wasm/src/transaction/utils.rs

```diff
@@ -0,0 +1,68 @@
+use cml_crypto_wasm::DatumHash;
+use wasm_bindgen::prelude::wasm_bindgen;
+use crate::{
+    transaction::{
+        TransactionOutput,
+        DatumOption,
+        ScriptRef,
+    },
+    NativeScript,
+    address::Address,
+    Value, Ed25519KeyHashList,
+};
+
+#[wasm_bindgen]
+impl TransactionOutput {
+    pub fn new(
+        address: &Address,
+        amount: &Value,
+        datum_option: Option<DatumOption>,
+        script_reference: Option<ScriptRef>
+    ) -> Self {
+        cml_chain::transaction::TransactionOutput::new(
+            address.clone().into(),
+            amount.clone().into(),
+            datum_option.map(Into::into),
+            script_reference.map(Into::into),
+        ).into()
+    }
+
+    pub fn address(&self) -> Address {
+        self.0.address().clone().into()
+    }
+
+    pub fn amount(&self) -> Value {
+        self.0.amount().clone().into()
+    }
+
+    pub fn set_amount(&mut self, amount: &Value) {
+        self.0.set_amount(amount.clone().into())
+    }
+
+    pub fn datum(&self) -> Option<DatumOption> {
+        self.0.datum().clone().map(Into::into)
+    }
+
+    /// Get the datum hash from a tx output if present as a hash.
+    /// Returns None if there is no datum, or the datum is inlined.
+    /// Use TransactionOutput::datum() for inlined datums.
+    pub fn datum_hash(&self) -> Option<DatumHash> {
+        self.0.datum_hash().cloned().map(Into::into)
+    }
+
+    pub fn script_ref(&self) -> Option<ScriptRef> {
+        self.0.script_ref().cloned().map(Into::into)
+    }
+}
+
+// TODO: anything here? pub type RequiredSignersSet = BTreeSet<Ed25519KeyHash>;
```

Comment:
you mean the question is whether we need an analogue to RequiredSignersSet? where will we use it, builders?